### PR TITLE
restructure tab to ipo and media block

### LINKED ISSUE #180: Review to Implement Tab Block into two Different Media and IPO Block

### PR DESCRIPTION
**Pull request by ravverma**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

- Ignore commented code, it is for showing the difference when API will return more then one IPO. I will set to API before merging
- Tabs block revert to the very first design and it is loading now media blocks inside.

Fix #180 

Test URLs:
- Before: https://main--icicidirect--aemsites.hlx.live/draft/ravverma/equity
- After: https://tab-ipo-media--icicidirect--aemsites.hlx.live/draft/ravverma/equity

- [ ] New Blocks introduced in this PR
      If yes, please provide details below

Block Name    | Documentation
------------- | -------------
Media | tab-ipo-media--icicidirect--aemsites.hlx.live/sidekick/blocks/media

Block Name    | Documentation
------------- | -------------
IPO | tab-ipo-media--icicidirect--aemsites.hlx.live/sidekick/blocks/ipo

### CODE REVIEW

**Comment by ravverma** [comment]
Media Authoring
image
IPO Authoring
image

File: blocks/media/media.js [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+  const moveDistance = (targetDotIndex - 1) * cardSize;
+  track.style.transform = `translateX(-${moveDistance}px)`;
+}
+
+async function createPicture(imageUrl, mediaWrapper) {
+  mediaWrapper.appendChild(createPictureElement(imageUrl, 'mqdefault', false));
+}
+
+function createMediaCards(container, data, apiKey, cardWidth) {
+  data.forEach((item) => {
```
Comment (anagarwa) [inline blocks/media/media.js:105]: suggestion , we can https://github.com/aemsites/icicidirect/blob/main/scripts/dom-builder.js here

File: blocks/ipo/ipo.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
+import {
+  createPictureElement, fetchData,
+  // eslint-disable-next-line no-unused-vars
```
Comment (anagarwa) [inline blocks/ipo/ipo.js:3]: do we need this ?

File: blocks/media/media.css [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+.block.media .title {
+    margin-bottom: 2.188rem;
+    position: relative;
+    text-align: center;
+    text-transform: uppercase;
+}
+
+.block.media .title::before {
+    content: "";
```
Comment (anagarwa) [inline blocks/media/media.css:19]: Do we need this property ?

File: blocks/media/media.css [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    background-size: 100%;
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    margin: -1.813rem 0 0 -1.813rem;
+    z-index: 10;
+}
+
+.block.media .card-info.videos-card .picture-wrapper::after {
+    background-image: url("../../icons/play-button.svg");
```
Comment (anagarwa) [inline blocks/media/media.css:182]: is it possible to use CreateIcon method from aem.js ?

File: blocks/ipo/ipo.js (hunk 2 above)
Comment (ravverma) [reply blocks/ipo/ipo.js:3]: yes , actually PR is feching data from mock to show difference.
will switch back to api before merge then these comment will be removed.

File: blocks/media/media.css (hunk 3 above)
Comment (ravverma) [reply blocks/media/media.css:19]: image
yes to create strikethrough line.

File: blocks/media/media.css (hunk 4 above)
Comment (ravverma) [reply blocks/media/media.css:182]: I tried but didn't work , as this is sudo element to create a overlay of play button on video thumbnail

File: blocks/media/media.js [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+      break;
+  }
+  startCarousel(block);
+}
+
+export default async function decorate(block) {
+  const blockConfig = readBlockConfig(block);
+  block.setAttribute('api-key', blockConfig.type);
+  const titleText = blockConfig.title.trim();
+  const discoverMoreButton = block.querySelector('.button-container');
```
[the remark this answers is not in the document]
Comment (ravverma) [reply blocks/media/media.js:245]: yes, done

**Review by anagarwa — APPROVED** [review]

**Pull request merged**